From 9b98480753d6e19254c5d76ca5320de753089ebd Mon Sep 17 00:00:00 2001
From: Manoharan Vijaya Raghavan <quic_mraghava@quicinc.com>
Date: Wed, 30 Oct 2024 10:10:12 +0530
Subject: [PATCH] ways overlap Configuring SCID 5, 6, 7 with capacity 128KB

SCID 5 ==> RXFILL, TXDESC (128KB)
SCID 1 ==> RXDESC & TXCMPL, UPIDX (128KB)
SCID 6 ==> WKK PCIE ATTACH 5G (128KB)
SCID 7 ==> WKK PCIE ATTACH 6G (128KB)
SCID 2 ==> APSS 768KB

Change-Id: I502582718660649a40e265791e25558ff128b08a
Signed-off-by: Manoharan Vijaya Raghavan <quic_mraghava@quicinc.com>
---
 drivers/soc/qcom/llcc-qcom.c       | 12 ++++++++----
 include/linux/soc/qcom/llcc-qcom.h |  6 ++++++
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/drivers/soc/qcom/llcc-qcom.c b/drivers/soc/qcom/llcc-qcom.c
index 2e1e1fb6d556..d71624459604 100644
--- a/drivers/soc/qcom/llcc-qcom.c
+++ b/drivers/soc/qcom/llcc-qcom.c
@@ -358,10 +358,14 @@ static const struct llcc_slice_config sm8550_data[] =  {
 };
 
 static const struct llcc_slice_config ipq5424_data[] =  {
-	/* SCID 1 - Descriptors (PPE, CUMAC) */
-	{LLCC_CPUSS,     1,  768, 1, 0,  0xFFFF, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, },
-	/* SCID 2 - SKB, SKB data (PPE, CUMAC, APSS Core 0,1,2) */
-	{LLCC_VIDSC0,    2,  256, 2, 1,  0xF000, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, },
+	/* SCID 1 - RXDESC & TXCMPL Descriptors (PPE) */
+	{LLCC_RXDESC,     1,   128, 3, 0,  0x0FFF, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, },
+	/* SCID 2 - SKB, SKB data (APSS Core 0,1,2) */
+	{LLCC_APSS,       2,   768, 1, 1,  0x0FFF, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, },
+	/* SCID 5 - RXFILL, TXDESC descriptors (PPE) */
+	{LLCC_RXFILL,     5,   128, 3, 0,  0x0FFF, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, },
+	{LLCC_PCIE5G,     6,   128, 3, 0,  0xC000, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, },
+	{LLCC_PCIE6G,     7,   128, 3, 0,  0x3000, 0x0,  0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, },
 };
 
 static const struct llcc_edac_reg_offset llcc_v1_edac_reg_offset = {
diff --git a/include/linux/soc/qcom/llcc-qcom.h b/include/linux/soc/qcom/llcc-qcom.h
index 93417ba1ead4..315d5b2b6e27 100644
--- a/include/linux/soc/qcom/llcc-qcom.h
+++ b/include/linux/soc/qcom/llcc-qcom.h
@@ -55,6 +55,12 @@
 #define LLCC_DISP_WB	 53
 #define LLCC_DISP_1	 54
 #define LLCC_VIDVSP	 64
+#define LLCC_RXDESC 1
+#define LLCC_APSS   2
+#define LLCC_RXFILL 5
+#define LLCC_PCIE5G 6
+#define LLCC_PCIE6G 7
+
 
 /**
  * struct llcc_slice_desc - Cache slice descriptor
-- 
2.34.1

